Search2DMatrix.java

```diff
@@ -0,0 +1,42 @@
+class Search2DMatrix {
+
+    // Time Complexity: O(log r + log c)
+    // where r -> no. of rows and c -> no. of columns in the matrix
+
+    public boolean searchMatrix(int[][] matrix, int target) {
+        if(matrix == null || matrix.length == 0 || matrix[0].length == 0)
```

Comment:
Good Job! Please add space complexities for your solutions and review other inline comments.